src/index.js

```diff
@@ -176,8 +174,14 @@ const CPK = class CPK {
         [
           this.cpkProvider.constructor.getContractAddress(this.multiSend),
           0,
-          this.cpkProvider.encodeMultiSendCalldata(this.multiSend, standardizedTxs),
-          CPK.DELEGATECALL,
+          this.cpkProvider.constructor.encodeMultiSendCallData({
+            web3: this.cpkProvider.web3,
```

Comment:
to be honest this looks much worse than before :/ (now we have to pass all the params in)

Reply:
If the cpk is used in a supported public network (mainnet, rinkeby, goerli or kovan), only this is needed:
- Using `web3`: ({ web3, transactions })
- Using `ethers`: ({ ethers, signer, transactions })

If the cpk is used in an unsupported network, `multilSendAddress` must be provided too.

Reply:
I liked you approach for the provider and would do something similar everywhere, e.g. have a network protbider interface and  encoder interface and then there is a web3 and a ethers version (probably all that could be one class and not 3)

Reply:
To be fair I need to look at this in an ide and not only mobile (i am missing the overview). But I am not a big fan of all the if branches that check the provider

Reply:
@rmeissner Do you mean to have `encodeMultiSendCallData` as a method on a CPKProvider instead of just as a static method, and then to document the CPKProvider?